libs/langchain-community/src/retrievers/bm25.ts

```diff
@@ -53,6 +53,13 @@ export class BM25Retriever extends BaseRetriever {
 
     scoredDocs.sort((a, b) => b.score - a.score);
 
-    return scoredDocs.slice(0, this.k).map((item) => item.document);
+    return scoredDocs.slice(0, this.k).map((item) => new Document({
+      ...item.document.id && { id: item.document.id },
+      pageContent: item.document.pageContent,
+      metadata: {
+        score: item.score,
+        ...item.document.metadata,
+      }
+    }));
```

Comment:
I would prefer to use the ranked document structure `{document: Document, score: number}`; however, I believe this change could disrupt client applications that rely on the existing structure. By adding the score to the metadata instead, we introduce the feature in a non-disruptive manner.

Reply:
Yes thanks for this!